c/serverStatusService.c

```diff
@@ -0,0 +1,266 @@
+static JsonObject* serverConfig = NULL;
+char productVersion[40];
+extern char **environ;
+
+typedef int EXSMFI();
+EXSMFI *smf_func;
+
+int installServerStatusService(HttpServer *server, JsonObject *serverSettings, char* productVer)
+{
+  HttpService *httpService = makeGeneratedService("REST_Service", "/server/agent/**");
+  httpService->authType = SERVICE_AUTH_NATIVE_WITH_SESSION_TOKEN;
+  httpService->serviceFunction = &serveStatus;
+  httpService->runInSubtask = TRUE;
+  httpService->doImpersonation = TRUE;
+  registerHttpService(server, httpService);
+  serverConfig = serverSettings;
+  memcpy(productVersion, productVer, 40);
+  return 0;
+}
+
+void respondWithServerConfig(HttpResponse *response){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartObject(out, "options");
+  jsonPrintObject(out, serverConfig);
+  jsonEndObject(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerRoutes(HttpResponse *response){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonStartArray(out, "links");
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/config");
+  jsonAddString(out, "rel", "config");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/log");
+  jsonAddString(out, "rel", "log");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/logLevels");
+  jsonAddString(out, "rel", "logLevels");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonStartObject(out, NULL);
+  jsonAddString(out, "href", "/server/agent/environment");
+  jsonAddString(out, "rel", "environment");
+  jsonAddString(out, "type", "GET");
+  jsonEndObject(out);
+  jsonEndArray(out);
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithLogLevels(HttpResponse *response){
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  JsonObject *logLevels = jsonObjectGetObject(serverConfig, "logLevels");
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  if(logLevels){
+    jsonPrintObject(out, logLevels);
+  }
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerEnvironment(HttpResponse *response){
+  /*Information about parameters for smc_func: https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.erbb700/smfp.htm#smfp*/
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  struct utsname unameRet;
+  uname(&unameRet);
+  char pid[64];
+  char ppid[64];
+  char dp[64];
```

Comment:
It's unclear from the context what `dp`, `mvs_u`, etc, are. Can you please add a little more descriptive names?